src/shared/dissect-image.c

```diff
@@ -2284,6 +2287,8 @@ int dissected_image_mount_and_warn(
         assert(m);
 
         r = dissected_image_mount(m, where, uid_shift, uid_range, userns_fd, flags);
+        if (r == -ENOTEMPTY)
+                return log_error_errno(r, "A mount point directory in the image is not empty.");
```

Comment:
I think this message could be improved. i.e. "Image contains an already populated directory where an empty directory to to use as mount point is expected."